tests/specs/lemmas.k

```diff
@@ -335,9 +335,10 @@ module LEMMAS-HASKELL [symbolic, kore]
 
     rule 0 <=Int lengthBytes ( _ ) => true [smt-lemma, simplification]
 
-    // TODO: why do we need an explicit symmetry rule for `==K`? should this not happend automatically?
```

Comment:
why do we need an explicit symmetry rule for `==K`? I tried combining these two rules into one (on line 343 below) to see if it works without the symmetry rule

Reply:
The generalization I tried (on line 343 in the commit) does not work. I fails for a proof in `tests/specs/functional/lemmas-spec.k`, but that one only. Perhaps there is a correct way to though?